### PR TITLE
fix breaking changes from transformers 2 to 4

### PR DESCRIPTION
**Pull request by Skhmt**

### CODE REVIEW

**Comment by MikkoMMM** [comment]
This does break support with transformers 2.3.0, but the question is: who cares? I mean it non-ironically. Is there such a non-negligible group for whom this would produce a major problem?

I tested this with Pythons 3.7 and 3.9 and it seems to be working with both.

**Comment by MikkoMMM** [comment]
This also seems to work with transformers==4.0.0.

**Comment by Meanmelter** [comment]
There is no reason to keep support for 2.3.0 - We may be able to optimize the generation better with it too. Based off someone in GodAI the one drawback they saw was that it generates all the logits then gives them to you, there might be a way to cancel early or mess with the process but not that I am aware of. You could pass `MAX_LENGTH `to the` generate` function, but it just subtracts from the context.

**Comment by Skhmt** [comment]
This PR no longer needs to exist as  #147 includes these fixes and more.
